### PR TITLE
Move backward compatibility checks for factors to config.cpp

### PR DESCRIPTION
**Pull request by snukky**
### Description
It seems that those checks need to be done after model config options are loaded from model.npz to take an effect.

Added dependencies: none

### How to test
As other factorized models.

### Checklist

- [x] I have tested the code manually
- [x] I have run regression tests
- [x] I have read and followed CONTRIBUTING.md
- [ ] I have updated CHANGELOG.md

### CODE REVIEW

File: src/common/config.cpp [hunk 1]
```diff
@@ -116,6 +116,21 @@ void Config::initialize(ConfigParser const& cp) {
     config_["tsv-fields"] = tsvFields;
   }
 
+  // ensures factors backward compability whilst keeping the more user friendly CLI
+  if(get<std::string>("lemma-dependency").empty()) {
```
Comment (snukky) [inline src/common/config.cpp:120]: @pedrodiascoelho Is there a way to check if the model has factors? If so, maybe this should be added to this condition here?

**Comment by snukky** [comment]
@pedrodiascoelho Feel free to merge it if it looks OK, this will be reviewed in the factors main PR in the marian-nmt/marian-dev anyway.

File: src/common/config.cpp (hunk 1 above)
Comment (pedrodiascoelho) [reply src/common/config.cpp:120]: The factors are never explicit set via the cli. They are triggered if you provide a vocab file with the sufix `.fsv`, so I guess the only way would be checking if any vocab file finishes with `.fsv`, although I'm not sure if this is a clean solution.

Comment (snukky) [reply src/common/config.cpp:120]: Thanks, then let's keep this as it is.

**Comment by pedrodiascoelho** [comment]
You were right, this way the change was not triggered if you loaded a pretrained model. Tested and it works!
Will merge it!

**Pull request closed without merging**